acceptable/openapi.py

```diff
@@ -22,32 +23,147 @@ def _to_dict(source: Any):
+                }
+            },
+            "responses": {
+                "200": {
+                    "description": "OK",
+                    "content": {
+                        "application/json": {
+                            "schema": {
+                                "$ref": "#/components/schemas/Default",
+                            }
+                        }
+                    },
+                },
+            },
+        }
+
+        if self.summary is None or self.summary == "":
+            result.pop("summary")
+
+        return result
+
+
 @dataclass
-class OasInfo(object):
+class OasPath:
+    operation: dict = field(default_factory=lambda: {})
+
+    def _to_dict(self):
+        return _to_dict(self.operation)
+
+
+@dataclass
+class OasInfo:
     description: str = ""
     version: str = ""
     title: str = ""
-    tags: list = field(default_factory=lambda: [])
-    contact: dict = field(default_factory=lambda: {"name": "", "email": ""})
+    contact: dict = field(
+        default_factory=lambda: {"name": "", "email": "example@example.example"}
+    )
 
 
 @dataclass
-class OasRoot31(object):
+class OasRoot31:
     openapi: str = "3.1.0"
     info: OasInfo = OasInfo()
-    servers: dict = field(default_factory=lambda: {})
+    tags: list = field(default_factory=lambda: [])
+    servers: list = field(default_factory=lambda: [{"url": "http://localhost"}])
     paths: dict = field(default_factory=lambda: {})
-    components_schemas: dict = field(default_factory=lambda: {})
 
+    def _to_dict(self):
+        return {
+            "openapi": self.openapi,
+            "info": _to_dict(self.info),
+            "servers": _to_dict(self.servers),
+            "tags": _to_dict(self.tags),
+            "paths": _to_dict(self.paths),
+            "components": {
+                "schemas": {
+                    "Default": {
+                        "type": "object",
+                        "properties": {
+                            "code": {
+                                "type": "integer",
+                                "format": "int32",
+                            }
+                        },
+                    }
+                }
+            },
+        }
 
-def dump(metadata: APIMetadata, stream=None):
-    service_name = None
-    if len(metadata.services) == 1:
-        service_name = list(metadata.services.keys())[0]
 
+def dump(metadata: APIMetadata, stream=None):
     oas = OasRoot31()
-    oas.info.title = service_name or ""
-    oas.info.version = metadata.current_version or ""
+    oas.info.title = (
+        get_head_of_single_item_list(list(metadata.services.keys()), "service")
+        or "None."
+    )
+    oas.info.description = oas.info.title
+    oas.info.version = "0.0." + str(metadata.current_version)
+    tags = set()
+
+    for _, service_group in metadata.services.items():
+        for _, api_group in service_group.items():
+            for _, endpoint in api_group.items():
```

Comment:
Maybe just use `.values()` if keys are being ignored in these 3 lines.

Reply:
Done.